extension/src/v0.rs

```diff
@@ -1,5 +1,5 @@
 #[cfg(test)]
```

Comment:
The introduction of versioning to this crate seems off, if the intended goal was a generic chain extension and this seems very specific. I guess part of this is to eliminate duplicate code, but from a modularity perspective I wonder if versioning should be a 'layer above'. 

I am probably missing the nuances here, so will take some time to consider it further...

Reply:
Created an issue: https://github.com/r0gue-io/pop-node/issues/180